cognite/neat/_utils/cdf/loaders/_data_modeling.py

```diff
@@ -206,45 +209,70 @@ def are_equal(self, local: ViewApply, remote: View) -> bool:
     def as_write(self, view: View) -> ViewApply:
         return ViewApply.load(self._as_write_raw(view))
 
-    def retrieve_all_ancestors(self, views: list[ViewId]) -> ViewList:
-        return ViewList(self._retrieve_view_ancestors(views, self._cache_view_by_id))
+    def retrieve_all_ancestors(self, views: list[ViewId], include_connections: bool = False) -> ViewList:
+        return ViewList(self._retrieve_view_ancestors(views, include_connections, self._cache_view_by_id))
 
-    def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, View]) -> list[View]:
+    def _retrieve_view_ancestors(
+        self, parents: list[ViewId], include_connections: bool, cache: dict[ViewId, View]
+    ) -> list[View]:
         """Retrieves all ancestors of a view.
 
         This will mutate the cache passed in, and return a list of views that are the ancestors
         of the views in the parents list.
 
         Args:
             parents: The parents of the view to retrieve all ancestors for
+            include_connections: Whether to include all sources.
             cache: The cache to store the views in
         """
-        parent_ids = parents.copy()
+        next_backlog_ids = parents.copy()
         found: list[View] = []
         found_ids: set[ViewId] = set()
-        while parent_ids:
+        while next_backlog_ids:
             to_lookup: set[ViewId] = set()
-            grand_parent_ids = []
-            for parent in parent_ids:
-                if parent in found_ids:
+            backlog_ids: list[ViewId] = []
+            for backlog_id in next_backlog_ids:
+                if backlog_id in found_ids:
                     continue
-                elif parent in cache:
-                    found.append(cache[parent])
-                    grand_parent_ids.extend(cache[parent].implements or [])
+                elif backlog_id in cache:
+                    parent_view = cache[backlog_id]
+                    found.append(parent_view)
+                    self._update_backlog(parent_view, backlog_ids, found_ids, include_connections)
                 else:
-                    to_lookup.add(parent)
+                    to_lookup.add(backlog_id)
 
             if to_lookup:
                 looked_up = self.client.data_modeling.views.retrieve(list(to_lookup))
                 cache.update({view.as_id(): view for view in looked_up})
                 found.extend(looked_up)
                 found_ids.update({view.as_id() for view in looked_up})
                 for view in looked_up:
-                    grand_parent_ids.extend(view.implements or [])
+                    self._update_backlog(view, backlog_ids, found_ids, include_connections)
 
-            parent_ids = grand_parent_ids
+            next_backlog_ids = backlog_ids
         return found
 
+    @staticmethod
+    def _update_backlog(
```

Comment:
should this be defined under `def _retrieve_view_ancestors` as "local" method? I do not see it being used elsewhere